Makefile

```diff
@@ -263,10 +263,19 @@ controller-gen: ## Download controller-gen locally if necessary.
 	@test '$(shell $(CONTROLLER_GEN) --version)' = 'Version: $(controller_gen_version)' ||\
 	$(call go-get-tool,sigs.k8s.io/controller-tools/cmd/controller-gen@$(controller_gen_version))
 
+.PHONY: kustomize
 KUSTOMIZE = $(shell pwd)/bin/kustomize
+KUSTOMIZE_VERSION = v4.5.7
 kustomize: ## Download kustomize locally if necessary.
-	@test -f $(KUSTOMIZE) ||\
-	$(call go-get-tool,sigs.k8s.io/kustomize/kustomize/v4@v4.5.7)
+	@test '$(shell $(KUSTOMIZE) version | tr -d "{" | tr -d "}" |  cut -d" " -f1)' = 'Version:kustomize/$(KUSTOMIZE_VERSION)' ||\
+	{ \
+	set -e ;\
+	mkdir -p $(dir $(KUSTOMIZE)) ;\
+	OS=$(shell go env GOOS) && ARCH=$(shell go env GOARCH) && \
+	curl -sSL https://github.com/kubernetes-sigs/kustomize/releases/download/kustomize%2F$(KUSTOMIZE_VERSION)/kustomize_$(KUSTOMIZE_VERSION)_$${OS}_$${ARCH}.tar.gz \
+	| tar xz -C $(dir $(KUSTOMIZE)) ;\
```

Comment:
If curl fail the pipe may succeed and we may end with broken tool or older version of the tool.

Reply:
ack